pkgs/tools/typesetting/tex/texlive/bin.nix

```diff
@@ -241,92 +245,73 @@ core-big = stdenv.mkDerivation { #TODO: upmendex
+      utils/{tpic2pdftex,vlna,xindy,xml2pmx,xpdfopen}
+    mkdir WorkDir
+    cd WorkDir
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform)
   # without this, the native builds attempt to use the binary
   # ${target-triple}-gcc, but we need to use the wrapper script.
   ''
     export BUILDCC=${buildPackages.stdenv.cc}/bin/cc
-  ''
-  +
-  ''
-    mkdir ./WorkDir && cd ./WorkDir
-    for path in libs/{pplib,teckit,lua53${luajit}} texk/web2c; do
-      (
-        if [[ "$path" =~ "libs/lua" ]]; then
-          extraConfig="--enable-static --disable-shared"
-        else
-          extraConfig=""
-        fi
-  '' + lib.optionalString (!stdenv.buildPlatform.canExecute stdenv.hostPlatform)
+  '';
+
+  configureFlags = common.configureFlags
+    ++ withSystemLibs [ "kpathsea" "ptexenc" "cairo" "harfbuzz" "icu" "graphite2" ]
+    ++ map (prog: "--disable-${prog}") # don't build things we already have
+      # list from texk/web2c/configure
+      ([ "tex" "ptex" "eptex" "uptex" "euptex" "aleph" "hitex" "pdftex"
+        "web-progs" "synctex"
+      ] ++ lib.optionals (!withLuaJIT) [ "luajittex" "luajithbtex" "mfluajit" ])
+    /* disable all packages, re-enable upmendex, web2c packages */
+    ++ [ "--disable-all-pkgs" "--enable-upmendex" "--enable-web2c" ]
+    /* kpathsea requires specifying the kpathsea location manually */
+    ++ [ "--with-kpathsea-includes=${core.dev}/include" ]
     # results of the tests performed by the configure scripts are
     # toolchain-dependent, so native components and cross components cannot use
     # the same cached test results.
     # Disable the caching for components with native subcomponents.
-  ''
-        if [[ "$path" =~ "libs/luajit" ]] || [[ "$path" =~ "texk/web2c" ]]; then
-          extraConfig="$extraConfig --cache-file=/dev/null"
-        fi
-  ''
-  +
-  ''
-        mkdir -p "$path" && cd "$path"
-        "../../../$path/configure" $configureFlags $extraConfig
+    ++ lib.optional (!stdenv.buildPlatform.canExecute stdenv.hostPlatform)
+      " --cache-file=/dev/null";
 
-        if [[ "$path" =~ "libs/luajit" ]] || [[ "$path" =~ "libs/pplib" ]]; then
-          # ../../../texk/web2c/mfluadir/luapeg/lpeg.h:29:10: fatal error: 'lua.h' file not found
-          # ../../../texk/web2c/luatexdir/luamd5/md5lib.c:197:10: fatal error: 'utilsha.h' file not found
-          make ''${enableParallelBuilding:+-j''${NIX_BUILD_CORES}}
-        fi
-      )
-    done
-  '';
+  configureScript = "../configure";
 
-  preBuild = "cd texk/web2c";
   enableParallelBuilding = true;
 
   doCheck = false; # fails
 
-  # now distribute stuff into outputs, roughly as upstream TL
-  # (uninteresting stuff remains in $out, typically duplicates from `core`)
   outputs = [
     "out"
+    "dev"
+    "man"
+    "info"
     "metafont"
     "mflua"
     "metapost"
     "luatex"
     "luahbtex"
-    "luajittex"
+    "ptex"
+    "uptex"
```

Comment:
there are currently overrides for ptex and uptex in tlpdb-overrides that make the bin containers look for some binaries in bin.metapost, but now the binaries are placed in bin.ptex / bin.uptex.

So the overrides can be removed:
https://github.com/NixOS/nixpkgs/blob/c372337790cee1194c0ee5430a90c4739eb39a5a/pkgs/tools/typesetting/tex/texlive/tlpdb-overrides.nix#L123-L149